gammapy/modeling/models/spatial.py

```diff
@@ -563,6 +565,8 @@ def __init__(
             filename = str(make_path(filename))
 
         self.map = map
+        if self.map.unit != "sr-1":
```

Comment:
I would just keep this check here. It's much better to introduce a convention and let the code fail early (on init of the `TemplateSpatialModel` ), I would modify the code to also allow equivalent units such as `deg2`. You can use e.g. `unit.is_equivalent()`. Also adapt the error message...